### PR TITLE
Reuse memberfield js

### PR DESCRIPTION
**Pull request by ivan-unfolds**
@rjmk again this is not ready to be merged in but there for you to take a look to see if we are sailing in right direction. We have not moved everything into shared just yet as we wanted to see whether everything worked first.

@rjmk update 09/08

This PR includes:
- Moving Mode and Member modules into shared.
- Implementing Redux Form on the member view.
- Reuse Rafe's field.js files.
- Reset form correctly on both admin and member's view.
- Initial styling of member's form for mobile view.

### CODE REVIEW

**Comment by ivan-unfolds** [comment]
@rjmk here is our latest work for your consideration.

File: src/shared/dumb_components/member_fields.js [hunk 1]
```diff
@@ -18,9 +15,10 @@ const PersonalFields = (
   , className
   , buttons_first
   , error
+  , read_only
   }
 ) => {
-  const fs = (fields.membership_type.value || '').match('life')
+  const fs = ((fields.membership_type && fields.membership_type.value) || '').match('life')
```
Comment (rjmk) [inline src/shared/dumb_components/member_fields.js:24]: :heart:

File: src/shared/form_fields/member.js [hunk 2]
```diff
@@ -105,6 +104,24 @@ const required =
   , 'news_type'
   ]
 
+const read_only_user =
```
Comment (rjmk) [inline src/shared/form_fields/member.js:143]: This is fine, though I would probably define it as `read_only.concat([ 'due_date' ])`

File: src/shared/redux/modules/member.js [hunk 3]
```diff
@@ -128,7 +127,9 @@ const to_errors = (dispatch) => ({ body: { invalidAttributes } }) => {
 
 const id_value = compose(String, path(['body', 'id']))
 
-const null_empty = map((v) => v == '' ? null : v)
+// TODO: WE CHANGED THIS TO TRIPLE EQUALS, IS THAT GOOD?
```
Comment (rjmk) [inline src/shared/redux/modules/member.js]: Sorry I thought I'd commented on this (perhaps in gitter) -- this is good

**Comment by ivan-unfolds** [comment]
@rjmk I have merged in the styling branch so this is ready for you to have a look at now.

**Comment by rjmk** [comment]
@ivanmauricio Great, thanks. Just so we can clarify for ourselves, could you update the PR description to summarise it?

File: src/shared/redux/modules/member.js [hunk 4]
```diff
@@ -155,6 +154,19 @@ export const update_member = createAction
...
 
+export const fetch_member_user = createAction
+  ( FETCHED_MEMBER
+  , () => compose(map(to_member), get_body)('/api/account')
+  )
+
+export const update_member_user = createAction
+  ( UPDATED_MEMBER
+  , (member, dispatch) => compose
+    ( map(errors_or_to_member(dispatch))
```
Comment (rjmk) [inline src/shared/redux/modules/member.js]: I would factor out the similarity to `update_member`

File: src/shared/redux/modules/member.js [hunk 5]
```diff
@@ -155,6 +154,19 @@ export const update_member = createAction
     )(member.id)
   )
 
+export const fetch_member_user = createAction
+  ( FETCHED_MEMBER
+  , () => compose(map(to_member), get_body)('/api/account')
```
Comment (rjmk) [inline src/shared/redux/modules/member.js]: I would factor out the similarity to `fetch_member`

**Comment by rjmk** [comment]
Looks great. A touch more refactoring to do in the members module

**Comment by ivan-unfolds** [comment]
@rjmk got a bit defeated by this refactor. Tried lots of stuff which didn't work so went for something simple in the end. Would love to know if there is a better way.

**Comment by rjmk** [comment]
@ivanmauricio Do these two functions suggest a better way?

`fetch_user_url = _ => '/api/account'`
`fetch_member_url = compose(make_user_url(get_user_url), prop('id'))`

**Comment by ivan-unfolds** [comment]
Thanks @rjmk, that solved all my problems 😄

**Comment by rjmk** [comment]
Great 👍

**Pull request merged**